app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -135,6 +139,10 @@ def start_vm(uuid)
       find_vm_by_uuid(uuid).start(:blocking => true)
     end
 
+    def start_with_cloudinit(uuid,user_data=nil)
```

Comment:
Modify `start_vm` instead of creating a new method, so that it can detect whether user_data was passed or not. As far as I see, the `vm` object you find should also contain the user data. `setUserData` in orchestration/compute runs before `setCompute`, which initializes the vm with compute_attributes (which contain the user data)